evalai/challenges.py

```diff
@@ -272,10 +276,54 @@ def submit(ctx, file, annotation, large, public, private):
                 submission_metadata["publication_url"] = click.prompt(
                     style("Publication URL", fg="yellow"), type=str, default=""
                 )
+            submission_meta_attributes = get_submission_meta_attributes(
+                ctx.challenge_id, ctx.phase_id)
+            submission_attribute_metadata = []
+            if submission_meta_attributes and len(submission_meta_attributes) > 0:
+                if click.confirm("Do you want to include the Submission Metadata"):
+                    for attribute in submission_meta_attributes:
+                        typ = attribute["type"]
+                        name = attribute["name"]
+                        desc = attribute["description"]
+                        value = None
+                        message = "{} ({})".format(name, desc)
+                        if(typ == "text"):
+                            value = click.prompt(
+                                style(message, fg="yellow"), type=str, default=""
+                            )
+                        if(typ == "boolean"):
+                            value = click.prompt(
+                                style(message, fg="yellow"), type=bool
+                            )
+                        if(typ == "radio"):
+                            value = click.prompt(
+                                style("{}: Choices:{}".format(message, attribute["options"]), fg="yellow"), type=click.Choice(attribute["options"])
+                            )
+                        if(typ == "checkbox"):
+                            optionChosen = True
+                            while optionChosen:
+                                value = []
+                                choices = click.prompt(
+                                    style("{}: Choices(0 or more separated by comma):{}".format(message, attribute["options"]), fg="yellow"), type=str
+                                )
+                                choices = choices.strip(' ').split(',')
+                                for choice in choices:
+                                    if choice in attribute['options']:
+                                        value.append(choice)
+                                        optionChosen = False
+                                    else:
+                                        echo(
+                                            "Error: Choose correct value(s) from the given options only")
+                                        optionChosen = True
```

Comment:
We also need to check atleast 1 value is added if the field is mandatory. If its option user can skip adding the field too. Can we please handle that as well

Reply:
How to know if the field is mandatory?
```json
[{"name": "TextAttribute", "type": "text", "description": "Sample"}, {"name": "SingleOptionAttribute", "type": "radio", "options": ["A", "B", "C"], "description": "Sample"}, {"name": "MultipleChoiceAttribute", "type": "checkbox", "options": ["alpha", "beta", "gamma"], "description": "Sample"}, {"name": "TrueFalseField", "type": "boolean", "description": "Sample"}]
```
This is the submission meta attributes. They give no information if the field is optional or important.
Currently, the user can select 0 or more options in the checkbox type field.

Reply:
@burnerlee hosts can pass `required` field when creating a challenge for each meta attribute. See [this](https://github.com/Cloud-CV/EvalAI-Starters/blob/master/challenge_config.yaml#L65) for an example

Reply:
It would be good to show a `*` for the field which is mandatory.